pallets/pallet-tx-pause/src/lib.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020-2021 Manta Network.
+// This file is part of Manta.
+//
+// Manta is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Manta is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Manta.  If not, see <http://www.gnu.org/licenses/>.
+//
+// The pallet-tx-pause pallet is forked from Acala's transaction-pause module https://github.com/AcalaNetwork/Acala/tree/master/modules/transaction-pause
+// The original license is the following - SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::unused_unit)]
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+use frame_support::{
+	dispatch::{CallMetadata, GetCallMetadata},
+	pallet_prelude::*,
+	traits::{Contains, PalletInfoAccess},
+	transactional,
+};
+use frame_system::pallet_prelude::*;
+use sp_runtime::DispatchResult;
+use sp_std::{prelude::*, vec::Vec};
+
+mod mock;
+mod tests;
+pub mod weights;
+
+pub use pallet::*;
+pub use weights::WeightInfo;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// The origin which may set filter.
+		type UpdateOrigin: EnsureOrigin<Self::Origin>;
+
+		/// Weight information for the extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// can not pause
+		CannotPause,
+		/// invalid character encoding
+		InvalidCharacter,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Paused transaction . \[pallet_name_bytes, function_name_bytes\]
+		TransactionPaused(Vec<u8>, Vec<u8>),
+		/// Unpaused transaction . \[pallet_name_bytes, function_name_bytes\]
+		TransactionUnpaused(Vec<u8>, Vec<u8>),
+	}
+
+	/// The paused transaction map
+	///
+	/// map (PalletNameBytes, FunctionNameBytes) => Option<()>
+	#[pallet::storage]
+	#[pallet::getter(fn paused_transactions)]
+	pub type PausedTransactions<T: Config> =
+		StorageMap<_, Twox64Concat, (Vec<u8>, Vec<u8>), (), OptionQuery>;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight(T::WeightInfo::pause_transaction())]
+		#[transactional]
+		pub fn pause_transaction(
```

Comment:
Any documentation on both calls? Because users will see the documentation while making transaction.
If we don't provide any documentation, users just see empty description about the call they choose.